packages/common/src/providers/httpProvider.ts

```diff
@@ -48,11 +51,18 @@ export class HttpProvider implements Provider {
             }
 
             const setCookieValue = res.headers.get('set-cookie');
-            if (setCookie && setCookieValue) {
-                setCookie(setCookieValue);
+            if (onSetCookie && setCookieValue) {
+                onSetCookie(setCookieValue);
+                if (autoUpdateCookie) {
+                    this.updateCookie(setCookieValue);
+                }
```

Comment:
So all this is supposed to happen only if onSetCookie is set? I would assume the following to be more correct:
```suggestion
            if (setCookieValue) {
                onSetCookie?.(setCookieValue);
                if (autoUpdateCookie) {
                    this.updateCookie(setCookieValue);
                }
```